datafusion/spark/src/function/datetime/make_dt_interval.rs

```diff
@@ -0,0 +1,479 @@
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::user_defined(Volatility::Immutable),
+        }
+    }
+}
+
+impl ScalarUDFImpl for SparkMakeDtInterval {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &str {
+        "make_dt_interval"
+    }
+
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
+        Ok(DataType::Duration(Microsecond))
+    }
+
+    fn invoke_with_args(&self, args: ScalarFunctionArgs) -> Result<ColumnarValue> {
+        if args.args.is_empty() {
+            return Ok(ColumnarValue::Scalar(ScalarValue::DurationMicrosecond(
+                Some(0),
+            )));
+        }
+        make_scalar_function(make_dt_interval_kernel, vec![])(&args.args)
+    }
+
+    fn coerce_types(&self, arg_types: &[DataType]) -> Result<Vec<DataType>> {
+        if arg_types.len() > 4 {
+            return exec_err!(
+                "make_dt_interval expects between 0 and 4 arguments, got {}",
+                arg_types.len()
+            );
+        }
+
+        Ok((0..arg_types.len())
+            .map(|i| {
+                if i == 3 {
+                    DataType::Float64
+                } else {
+                    DataType::Int32
+                }
+            })
+            .collect())
+    }
+}
+
+fn make_dt_interval_kernel(args: &[ArrayRef]) -> Result<ArrayRef, DataFusionError> {
+    let n_rows = args[0].len();
+    let days = args[0]
+        .as_primitive_opt::<Int32Type>()
+        .ok_or_else(|| plan_datafusion_err!("make_dt_interval arg[0] must be Int32"))?;
+    let hours: Option<&PrimitiveArray<Int32Type>> = args
+        .get(1)
+        .map(|a| {
+            a.as_primitive_opt::<Int32Type>().ok_or_else(|| {
+                plan_datafusion_err!("make_dt_interval arg[1] must be Int32")
+            })
+        })
+        .transpose()?;
+    let mins: Option<&PrimitiveArray<Int32Type>> = args
+        .get(2)
+        .map(|a| {
+            a.as_primitive_opt::<Int32Type>().ok_or_else(|| {
+                plan_datafusion_err!("make_dt_interval arg[2] must be Int32")
+            })
+        })
+        .transpose()?;
+    let secs: Option<&PrimitiveArray<Float64Type>> = args
+        .get(3)
+        .map(|a| {
+            a.as_primitive_opt::<Float64Type>().ok_or_else(|| {
+                plan_datafusion_err!("make_dt_interval arg[3] must be Float64")
+            })
+        })
+        .transpose()?;
+    let mut builder = DurationMicrosecondBuilder::with_capacity(n_rows);
+
+    for i in 0..n_rows {
+        // if one column is NULL → result NULL
+        let any_null_present = days.is_null(i)
+            || hours.as_ref().is_some_and(|a| a.is_null(i))
+            || mins.as_ref().is_some_and(|a| a.is_null(i))
+            || secs.as_ref().is_some_and(|a| {
+                a.is_null(i) || a.value(i).is_infinite() || a.value(i).is_nan()
```

Comment:
```suggestion
                a.is_null(i) || !a.value(i).is_finite()
```

Minor simplification